src/muxer/yamux/yamux_stream.cpp

```diff
@@ -195,30 +195,32 @@ namespace libp2p::connection {
     if (!is_writable_) {
       return cb(Error::NOT_WRITABLE);
     }
+    std::vector<uint8_t> in_vector(in.size());
+    std::copy(in.begin(), in.end(), in_vector.begin());
```

Comment:
`in` can is used in 3 branches:
- push queue;
- first sync write;
- first sync write and second async write.
Copy to `vector` is required only in 2 cases:
- push queue (vector conversion can be moved into `if (is_writing_)`);
- async write (consider making `in` arg of write_lambda, then capture vector in `streamOnWindowUpdate` callback).